### DISCUSSION TITLE
support  geosite for outbound rule

### DISCUSSION THREAD

**Discussion by yaser-k** [mosajjal/sniproxy#54] [Ideas]
supporting geosite for outbound rule make it much easier to manage which service go through proxy.
https://github.com/v2fly/domain-list-community

**Comment by mosajjal** [comment]
need to answer a few questions before even thinking about this. nowadays, most of the web is sitting behind CDNs (Cloudfront, Cloudflare etc), which are not easy to geolocate. if we only go by geolocation, you could be hitting a Bahrain AWS IP which happens to be your closest termination point at the time. does that mean we block that? or allow that? 

This is not to mention anycast IPs. where is 1.1.1.1? geolocation shows it as Australia but it's actually distributed. 

This is not as big of a problem for source IPs since residential IP ranges are not as fluid. Any ideas/thoughts?

**Reply by yaser-k** [reply]
you are all right if we talk about geoip. but **geosite** is a list of domain names categorized in services. and support wildcard and regex.
take a look at https://github.com/v2fly/domain-list-community/tree/master/data and you will see what I mean. 
and I think it should match with inbound SNI that client send.
